### DISCUSSION TITLE
How to fix "Permission denied to access object"?

### DISCUSSION THREAD

**Discussion by GrayFace** [Tampermonkey/tampermonkey#2287] [Q&A] [answered]
I need to make a function in a script available to be called from console. I've added it to unsafeWindow, but when trying to call it I get "Permission denied to access object".

**Comment by derjanb** [comment, marked as answer]
Please take a look at [`cloneInto`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Sharing_objects_with_page_scripts#cloneinto).

**Reply by GrayFace** [reply]
> Please take a look at [`cloneInto`](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Sharing_objects_with_page_scripts#cloneinto).

Thank you!

Here's my code for whoever may need it: `unsafeWindow.Name = exportFunction(Name, unsafeWindow); // same as cloneInto(Name, unsafeWindow, {cloneFunctions: true});`
